src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -310,8 +311,8 @@ public boolean initiateConnection(Socket sock, Long sid) {
      * possible long value to lose the challenge.
      *
      */
-    public void receiveConnection(Socket sock) {
-        Long sid = null, protocolVersion = null;
+    public void receiveConnection(Socket sock) throws ConfigException {
+        Long sid, protocolVersion;
```

Comment:
As we removed the extraneous `null` here we could also use the primitive long type, right? I don't see any gain of using boxed types here. I see a lot of boxing/unboxing down in this method, so it could be simply `int sid, protocolVersion;` (this slightly falls into "the don't fix if..." adage, but we can ignore this for once ;) ).